CMakeLists.txt

```diff
@@ -39,9 +45,21 @@ message(status " CMAKE_CXX_FLAGS_DEBUG: ${CMAKE_CXX_FLAGS_DEBUG}")
 message(status " CMAKE_CXX_FLAGS_RELEASE: ${CMAKE_CXX_FLAGS_RELEASE}")
 
 if(Boost_FOUND AND OpenCV_FOUND)
+    
     ADD_DEFINITIONS("-DBOOST_LOG_DYN_LINK")
     include_directories(${Boost_INCLUDE_DIRS})
-    include_directories(${OpenCV_INCLUDE_DIRS})
-    add_executable(warg-cv main.cpp)
-    target_link_libraries(warg-cv ${OpenCV_LIBS} ${Boost_LIBRARIES} Core ReportGenerator CSVCreator TargetAnalysis TargetIdentification VolumeCalculation ImageImport GoogleMapsMatcher)
+    include_directories(${OpenCV_INCLUDE_DIRS}) 
+
+    if(DeckLinkSDK_FOUND)
+        include_directories(${DeckLinkSDK_INCLUDE_PATH})    	
+        ADD_DEFINITIONS("-DHAS_DECKLINK")
+	SET(COMPILE_DEFINITIONS "${DeckLinkSDK_DEF}")
+    else(DeckLinkSDK_FOUND)
+	message(status "DeckLinkSDK not found. Omitting DeckLink features.")
+    endif(DeckLinkSDK_FOUND)
+
+    add_executable(warg-cv main.cpp ${DeckLinkSDK_SRC_FILE})
+    target_link_libraries(warg-cv ${OpenCV_LIBS} ${Boost_LIBRARIES} ${DeckLinkSDK_LIBS} Core ReportGenerator CSVCreator TargetAnalysis TargetIdentification VolumeCalculation ImageImport GoogleMapsMatcher)
```

Comment:
Similar to above comment. Also, note that OpenCV no longer needs to be here, it's just that boost is needed for logging, program options and multithreading. You only need to link libraries used in main.cpp here, since if a library is used in a submodule library it is already linked to that library.